kebechet/managers/thoth_advise/thoth_advise.py

```diff
@@ -0,0 +1,149 @@
+from kebechet.source_management import Issue
+from kebechet.source_management import MergeRequest
+from kebechet.utils import cloned_repo
+
+
+_LOGGER = logging.getLogger(__name__)
+
+_BRANCH_NAME = "kebechet_thoth"
+
+
+class ThothAdviseManager(ManagerBase):
+    """Manage updates of dependencies using Thoth."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize ThothAdvise manager."""
+        # TODO: What needs to be changed for event driven Kebechet??
+        # We do API calls once for merge requests and we cache them for later use.
+        self._cached_merge_requests = None
+        super().__init__(*args, **kwargs)
+
+    @property
+    def sha(self):
+        """Get SHA of the current head commit."""
+        return self.repo.head.commit.hexsha
+
+    def _construct_branch_name(self) -> str:
+        """Construct branch name for the updated dependency."""
+        return f"{_BRANCH_NAME}-{self.sha}"
+
+    def _git_push(
+        self, commit_msg: str, branch_name: str, files: list, force_push: bool = False
+    ) -> None:
+        """Perform git push after adding files and giving a commit message."""
+        self.repo.index.add(files)
+        self.repo.index.commit(commit_msg)
+        self.repo.remote().push(branch_name, force=force_push)
+
+    def _open_merge_request(
+        self, branch_name: str, labels: list, files: list
+    ) -> typing.Optional[int]:
+        """Open a pull/merge request for dependency update."""
+        commit_msg = "Auto generated update"
+        body = "Pipfile.lock updated by kebechet-thoth manager"
+
+        # Delete branch if it didn't change Pipfile.lock
+        diff = self.repo.git.diff("master", files)
+        if diff == "":
+            _LOGGER.info("No changes necessary, exiting...")
+            return
+
+        # push force always to keep branch up2date with the recent master and avoid merge conflicts.
+        _LOGGER.info('Pushing changes')
+        self._git_push(":pushpin: " + commit_msg, branch_name, files, force_push=True)
+
+        # Check if the merge request already exists
+        for mr in self._cached_merge_requests:
+            if mr.head_branch_name == branch_name:
+                _LOGGER.info('Merge request already exists, updating...')
+                return
+
+        _LOGGER.info('Opening merge request')
+        merge_request = self.sm.open_merge_request(
+            commit_msg, branch_name, body, labels
+        )
+        return merge_request
+
+    @staticmethod
+    def _write_advise(adv_results: list):
+        lock_info = adv_results[0]["report"][0][1]["requirements_locked"]
+        with open("Pipfile.lock", "w+") as f:
+            _LOGGER.info('Writing to Pipfile.lock')
+            _LOGGER.debug(f"{json.dumps(lock_info)}")
+            f.write(json.dumps(lock_info))
+
+    def _issue_advise_error(self, adv_results: list, labels: list):
+        """Create an issue if advise fails."""
+        error_info = adv_results[0]["report"][0][0][0]
```

Comment:
This array can hold multiple "reasonings". Can it be expanded in the issue body?

Reply:
Yeah, didn't realize this was the case